soundbay/utils/logging.py

```diff
@@ -125,11 +128,43 @@ def calc_metrics(self, epoch: int, mode: str = 'train', label_names: List[str] =
             wandb.log({f'{mode}_charts/conf_mat': wandb.plot.confusion_matrix(probs=None, y_true=self.label_list,
                                                                               preds=self.pred_list,
                                                                               class_names=label_names)},
-                                                                              step=epoch)
+                                                                              step=epoch, commit=True)
         self.pred_list = []  # flush
         self.label_list = []
         self.pred_proba_list = []
 
+    def upload_artifacts(self, audio: torch.Tensor, label: torch.Tensor, raw_wav: torch.Tensor, idx: torch.Tensor, sample_rate: int=16000, flag: str='train', epoch: int=1):
+        """upload algorithm artifacts to W&B during training session"""
+        matplotlib.use('Agg')
+        idx = idx.detach().cpu().numpy()
+        label = label.detach().cpu().numpy()
+
+
+        # Original wavs batch
+
+        artifact_wav = torch.squeeze(raw_wav).detach().cpu().numpy() * 100
```

Comment:
Think there could be a simple solution for normalization here - bring audio power up to "volume" dB, with X an argument.

This can be something like

```
artifact_wav = torch.squeeze(raw_wav).detach().cpu().numpy()
artifact_wav = artifact_wav * 10 ^ (volume / (20 * log10(np.std(artifact_wav ))))
```

Reply:
or even normalize it always to -6dB, something like:
`artifact_wav = torch.squeeze(raw_wav).detach().cpu().numpy() / (wav_wav.abs().max()) * 0.5`
the current implementation might introduce clipping in some wavs

Reply:
Well it was very confusing.
@noamzbr I think you are missing `log10(1/ np.std(artifact_wav )` in the lower signal expression.
Otherwise I have to put in a negative volume.
Had to catch up with your overwhelming acoustics experience xD